### PR TITLE
inject port and env from start script

### PR DESCRIPTION
**Pull request by tungv**

### CODE REVIEW

**Comment by tungv** [comment]
rebased to latest master

**Comment by nqbao** [comment]
can we use parameter instead of bash argument?

something like `npm start --port=3000 --env=development`, by default it will be port 3000 and dev environment?

File: scripts/start [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export NODE_ENV=$1
+export PORT=$2
+node --harmony build/server.js
```
Comment (nqbao) [inline scripts/start]: need to make sure current directory is expected, we can use something like below

```
cd `dirname $0`/ >/dev/null 2>&1
DIR=`pwd`
echo "pwd is ${DIR}"
```

**Comment by tungv** [comment]
so it must be `npm start -- --port=3000 --env=development`. does it look weird?

File: scripts/start (hunk 1 above)
Comment (tungv) [inline scripts/start]: because we will run `npm start` anyway so it will always the folder in which `package.json` is located

**Comment by hung-phan** [comment]
Can we make it like `npm start NODE_ENV=development`?

File: scripts/start [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export NODE_ENV=$1
```
Comment (nqbao) [inline scripts/start]: will also need a shebang here, ether bash or sh

**Comment by tungv** [comment]
possible!

File: scripts/start [hunk 3]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env bash
+export NODE_ENV=${1:-development}
+export PORT=${1:-3000}
```
Comment (nqbao) [inline scripts/start]: Should be 2 to take the 2nd parameter

**Comment by nqbao** [comment]
@tungv we will need to change the start script to 

```
if [[ -z "${NODE_ENV}" ]]; then
  export NODE_ENV=${1:-development}
fi

if [[ -z "${PORT}" ]]; then
  export PORT=${2:-3000}
fi

node --harmony build/server.js
```

so that we can use `NODE_ENV=production npm start`, then we are good to go.

**Comment by tungv** [comment]
rebased

**Comment by nqbao** [comment]
lgtm

**Comment by hung-phan** [comment]
:+1:

**Pull request merged**